internal/controller/credentials_manager.go

```diff
@@ -73,14 +74,32 @@ func (am *CredentialsManager) Onboard(ctx context.Context, TlsSkipVerify bool, O
 
 	// Set the appropriate headers
 	req.Header.Set("Content-Type", "application/json")
-
 	resp, err := client.Do(req)
 	if err != nil {
 		am.Logger.Error(err, "Unable to send post request")
 		return responseData, err
 	}
-	defer resp.Body.Close()
 
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// 200 OK - Continue processing the response
+	case http.StatusUnauthorized:
+		// 401 Unauthorized
+		err := errors.New("unauthorized: invalid credentials")
+		am.Logger.Error(err, "Received 401 Unauthorized")
```

Comment:
@aaronIllumio
I missed all those calls to `Logger.Error` in your PR https://github.com/illumio/cloud-operator/pull/31.
The signature is different between `logr` and `zap`. The signature in `zap` takes the string as the first parameter:
https://github.com/uber-go/zap/blob/master/logger.go#L261
So each call like:
```
Logger.Error(err, "...")
```
must be modified into:
```
Logger.Error("...", zap.Error(err))
```
or, if it's a suggared logger:
```
Logger.Errorw("...", "error", err)
```
Please fix this in another PR.

Reply:
@evanj80-illumio please at least fix the logs that you modify or add in this PR.

Reply:
Yep sounds good my logs are done